### PR TITLE
Youtube: Add tests

### PR DESCRIPTION
**Pull request by mia-pi-git**
caught a few bugs with this so, that was nice.

### CODE REVIEW

**Comment by mia-pi-git** [comment]
Is there a better way to handle the lack of API key beyond just returning if it doesn't exist? 
I didn't consider that.

**Comment by Zarel** [comment]
idk if I want active network-dependent tests. They should probably at least be skipped by default (with `it.skip` or something?)

**Comment by Zarel** [comment]
The alternative is mocking, but that's probably excessive.

**Comment by Zarel** [comment]
That CI failure is very weird to me. Do you know why it's trying to get `modlog_r/relationshipadvice.txt`?

**Comment by mia-pi-git** [comment]
I don't - nothing i did should relate to that. 
I don't understand why it's never failed before, if this happens when creating a room for the test.

**Comment by AnnikaCodes** [comment]
> I don't - nothing i did should relate to that.
> I don't understand why it's never failed before, if this happens when creating a room for the test.

I would expect `Config.nofswriting` to prevent this?

File: test/server/chat-plugins/youtube.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/**
+ * Tests for the Youtube room plugin.
+ * Written by mia-pi.
+ */
+'use strict';
+const YoutubeInterface = require('../../../.server-dist/chat-plugins/youtube').YoutubeInterface;
+const assert = require('../../assert').strict;
+const Config = require('../../../.server-dist/config-loader').load();
```
Comment (AnnikaCodes) [inline test/server/chat-plugins/youtube.js:8]:
```suggestion
const assert = require('../../assert').strict;
```
Config is already set up in `test/main.js`, so requiring it again is unnecessary, and moreover resets `Config.nofswriting`, which caused the tests to fail.

File: test/server/chat-plugins/youtube.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Tests for the Youtube room plugin.
+ * Written by mia-pi.
+ */
+'use strict';
+const YoutubeInterface = require('../../../.server-dist/chat-plugins/youtube').YoutubeInterface;
+const assert = require('../../assert').strict;
```
Comment (Zarel) [inline test/server/chat-plugins/youtube.js:7]:
```suggestion
const assert = require('../../assert');
```

I'm surprised `.strict` exists at all, but it clearly shouldn't be used here. Our assert library is already strict.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**